pkg/targetgroupbinding/resource_manager.go

```diff
@@ -325,6 +330,34 @@ func (m *defaultResourceManager) updateTargetHealthPodConditionForPod(ctx contex
 	return needFurtherProbe, nil
 }
 
+// updatePodAsHealthyForDeletedTGB updates pod's targetHealth condition as healthy when deleting a TGB
+// if the pod has readiness Gate.
+func (m *defaultResourceManager) updatePodAsHealthyForDeletedTGB(ctx context.Context, tgb *elbv2api.TargetGroupBinding) error {
+	targetHealthCondType := BuildTargetHealthPodConditionType(tgb)
+
+	allPodKeys := m.podInfoRepo.ListKeys(ctx)
+	for _, podKey := range allPodKeys {
+		pod, exists, err := m.podInfoRepo.Get(ctx, podKey)
```

Comment:
we should check for pod's namespace here as well. 
Since in theory you can have two targetGroupBinding in two namespace.

In practice, with our default Ingress/Service implementation, the TGB's name is same as TargetGroup's name and will be unique across namespaces.
but this will impact user created targetGroupBinding.

we can fix it in future versions as the impact is minimal